packages/headless/src/features/search/search-slice.ts

```diff
@@ -90,6 +91,9 @@ export const searchReducer = createReducer(
       state.response.facets = action.payload.response.facets;
       state.response.searchUid = action.payload.response.searchUid;
     });
+    builder.addCase(fetchInstantResults.fulfilled, (state, action) => {
+      state.response.searchUid = action.payload.searchUid;
+    });
```

Comment:
If we use instant results and then right click a result's title, is it associated with the current search or the instant results search?

Reply:
Hmm it always seems to get the `search.searchResponseId` value actually, which stems from the "true search".
This PR fixes the search event ID but not the click event associated ID, damn

Reply:
We might need to store a separate id for instant results and leverage it in the analytics, I don't know how complex this can get.

Reply:
https://coveord.atlassian.net/browse/KIT-2254

Let's get this merged, then I'll attack this one